app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ClientServiceError < StandardError; end
+
+class SearchKeywordJob < ApplicationJob
+  queue_as :default
+
+  def perform(search_result)
```

Comment:
Since Redis serializes objects for background jobs a best practice is to only store the record ID instead of the whole object:

```suggestion
  def perform(search_result_id)
```

Then the job should fetch the correspond record from the database.